### DISCUSSION TITLE
multiValueQueryStringParameters typing

### DISCUSSION THREAD

**Discussion by gregggilbert-ivadolabs** [aws-powertools/powertools-lambda-python#3640] [Q&A]
The current way to type query parameters is by using the `Annotated[str, Query()]` types, but what happens when we need `multiValueQueryStringParameters` such as a list of strings? 

By using `param_1: Annotated[list[str], Query]`, Swagger UI will reflect it correctly, but when you "Try It Out", the resulting query would be decoupled in the URL (e.g., `http://abc?param_1=value_1&param_1=value_2`. This will raise an error because only one of the `param_1` values will be retained by PowerTools' `Query()`, and an error will be raised saying that `param_1` must be a list. 

Any idea how `multiValueQueryStringParameters` can be handled and typed?

**Comment by leandrodamascena** [comment]
Hey @gregggilbert-ivadolabs! We don't support this right now, but we have an issue opened to implement it: https://github.com/aws-powertools/powertools-lambda-python/issues/3622

We hope to implement it in future releases.

Thanks

**Reply by gregggilbert-ivadolabs** [reply]
Note that when splitting endpoints in multiple files using Routers, the router.current_event.multi_value_query_string_parameters is unavailable. Only available in the APIGatewayRestResolver.current_event. That should be considered in the open issue scope.

**Comment by leandrodamascena** [comment]
I'm closing this discussion as resolved. Please reopen if you need.
